src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -71,7 +71,77 @@ void setConversationStore(ConversationStore conversationStore) {
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
     List<Conversation> conversations = conversationStore.getAllConversations();
+    String username = (String)request.getSession().getAttribute("user");
+
+    User user = userStore.getUser(username);
+
+    if (user == null) {
+      // user was not found, don't let them create a conversation
+      System.out.println("User not found:line 80  " + username);
+      response.sendRedirect("/profiles");
+      return;
+    }
+
     request.setAttribute("conversations", conversations);
+
+    String status = user.getStatus();
+
+    System.out.println("User RELOADED STATUS: " + status);
+    if (status != null)
+    request.setAttribute("status_name", status);
+
     request.getRequestDispatcher("/WEB-INF/view/profiles.jsp").forward(request, response);
     }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response)
+        throws IOException, ServletException {
+
+      String username = (String) request.getSession().getAttribute("user");
+      if (username == null) {
+        // user is not logged in, don't let them create a conversation
+        response.sendRedirect("/profiles");
+        return;
+      }
+
+      User user = userStore.getUser(username);
+      if (user == null) {
+        // user was not found, don't let them create a conversation
```

Comment:
Maybe change the comments here to say "dont let them view their profile" instead of "create a converstation"